reference/format/datagrid.rst

```diff
@@ -0,0 +1,166 @@
+Datagrid
+========
```

Comment:
I'd propose to add simple datagrid example to show whole configuration, e.g. this one https://github.com/orocrm/platform/blob/a329ce2efa09447d9260f9ca7788a74f553e30fb/src/Oro/Bundle/UserBundle/Resources/config/datagrid.yml#L134-L191